Files/DataModels/NavigationControlItems/DriveItem.cs

```diff
@@ -139,26 +135,21 @@ public static async Task<DriveItem> CreateFromPropertiesAsync(StorageFolder root
         {
             var item = new DriveItem();
 
-            await CoreApplication.MainView.DispatcherQueue.EnqueueAsync(async () => await item.SetBitmapImage(imageStream));
+            if (imageStream != null)
+            {
+                item.ThumbnailData = await imageStream.ToByteArrayAsync();
+            }
+
             item.Text = root.DisplayName;
             item.Type = type;
             item.Path = string.IsNullOrEmpty(root.Path) ? $"\\\\?\\{root.Name}\\" : root.Path;
             item.DeviceID = deviceId;
             item.Root = root;
-            await CoreApplication.MainView.DispatcherQueue.EnqueueAsync(() => item.UpdatePropertiesAsync());
```

Comment:
Aren't MaxSpace, FreeSpace, etc bound to UI? Does it work without `EnqueueAsync`?

Reply:
@gave92 I always suspected there wouldn't be any threading issue even if the property is bound to UI elsewhere. I don't think property change notifications are dependent on the UI thread, but I could be wrong.

This is absolutely something worth verifying.